src/index.js

```diff
@@ -183,7 +183,11 @@ const userEvent = {
     };
     const opts = Object.assign(defaultOpts, userOpts);
 
-    const computedText = text.slice(0, element.maxLength || text.length);
+    const maxLength =
+      element.maxLength && element.maxLength >= 0
```

Comment:
`element.maxLength && element.maxLength >= 0` seems slightly misleading because `>= 0` makes it look like `element.maxLength` will be applied if it's `0`, which isn't true. Particularly relevant in that jsdom seems to use 0 as its default/unlimited maxlength value, which deviates from specs/browser implementations.

Reply:
That's a really good point that I overlooked. I've updated this.

Reply:
For context, I opened https://github.com/jsdom/jsdom/issues/2927 since that's the real source of deviation